project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/execution/comparable/StacktraceComparator.kt

```diff
@@ -0,0 +1,5 @@
+package org.plan.research.minimization.plugin.execution.comparable
+
+interface StacktraceComparator {
+    fun areEqual(stack1: String, stack2: String): Boolean
+}
```

Comment:
Just extend ExceptionComparator I suppose

Reply:
I did it that way because it is not clear how to compare stacktraces with it
`interface ExceptionComparator {
    fun areEquals(exception1: CompilationException, exception2: CompilationException): Boolean
}`

CompilationException doesn't have any fields and I am not sure how to do anything meaningfull.

The only Idea I have is to write something like this

`override fun areEquals(exception1: CompilationException, exception2: CompilationException): Boolean {
        return when {
            exception1 is KotlincException.BackendCompilerException && exception2 is KotlincException.BackendCompilerException -> {
                TODO(compare)
            }
            exception1 is KotlincException.GenericInternalCompilerException && exception2 is KotlincException.GenericInternalCompilerException -> {
                TODO(compare)
            }
            ...`

Reply:
Something like this. I can think about it more, if you want. But my general idea was to make some abstract class-wrapper that would be able to handle the stack traces

```kotlin

abstract class StacktraceFirstComparator : ExceptionComparator {
    protected abstract val backedComparator: ExceptionComparator
    abstract fun areEquals(exception1: CompilationExceptionWithStacktrace, exception2: CompilationExceptionWithStacktrace): Boolean
    final override fun areEquals(exception1: CompilationException, exception2: CompilationException): Boolean {
        if (exception1 !is ExceptionWithStacktrace || exception2 !is ExceptionWithStacktrace) {
            return backedComparator.areEquals(exception1, exception2)
        }
        return areEquals(exception1, exception2)
    }
}
```

but now I've realised that we don't have any marker of exception with stacktrace. So you can just compare them by their type as you suggested or add a new abstract class for `CompilationExceptionWithStacktrace`